### PR TITLE
LUT-27811 : Newsletter creation: the “manage generation” functionalit…

### PR DESCRIPTION
**Pull request by eriatataserp00**
https://dev.lutece.paris.fr/bugtracker/issues/27811

# Upgrade Informations 
With the actual version of newsletter, models of templates and picture with templates are stored in ./target as files following the properties path
`
newsletter.path.image.newsletter.template=/themes/admin/shared/plugins/newsletter/images/newsletter_templates
newsletter.path.file.newsletter.template=/skin/plugins/newsletter/templates
`
They can be lost during compilation, especialy using mvn clean

With this upgrade there are gonna stored using IFileStoreServiceProvider using `newsletter.FileProviderStore=newsletterDatabaseFileProvider`
Template files are still read but if modified or added there are gonna be stored using IFileStoreServiceProvider 
If you want templates model or pictures associated with templates to save : 

- upgrade the repo : git pull

 
- put the files within ./webapp folder. 
  cp pathToProductionWebSite/target/lutece/WEB-INF/templates/skin/plugins/newsletter/templates/* PathToTheNewsletterPlugin/webapp/WEB-INF/templates/skin/plugins/newsletter/templates
  cp pathToProductionWebSite/target/lutece/themes/admin/shared/plugins/newsletter/images/newsletter_templates PathToTheNewsletterPlugin/webapp/themes/admin/shared/plugins/newsletter/images/newsletter_templates
  
- recompile the plugin and then the website
 
Then your file are gonna be read. If there are modified(or added) they will be stored in the database using IFileStoreServiceProvider and from that they are not gonna be erased anymore during compilation

### CODE REVIEW

File: src/java/fr/paris/lutece/plugins/newsletter/web/NewsletterJspBean.java [hunk 1]
```diff
@@ -597,6 +596,8 @@ public String getComposeNewsLetter( HttpServletRequest request )
 
         String strBaseUrl = AppPathService.getProdUrl( request );
 
+        // ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
```
Comment (FrancoisEricMerlin) [inline src/java/fr/paris/lutece/plugins/newsletter/web/NewsletterJspBean.java]: Delete this comment

**Comment by FrancoisEricMerlin** [comment]
Have you manage the possibility to continue to use template stored in file system

File: src/java/fr/paris/lutece/plugins/newsletter/web/NewsletterJspBean.java (hunk 1 above)
Comment (eriatataserp00) [reply src/java/fr/paris/lutece/plugins/newsletter/web/NewsletterJspBean.java]: Ok, I have deleted this comment

**Comment by eriatataserp00** [comment]
> Have you manage the possibility to continue to use template stored in file system

Yes, they can be read until those files are modified, then they are stored with IFileStoreProvider (of newsletter) it could configured. 
The reason so this ticket is the locacion of those files, that are ereased during compilation.
It is explained in the description of the pr.

**Pull request merged**